src/java.base/share/classes/java/lang/foreign/SegmentAllocator.java

```diff
@@ -257,89 +257,162 @@ default MemorySegment allocateFrom(AddressLayout layout, MemorySegment value) {
         return segment;
     }
 
+    /**
+     * Allocates a memory segment with the given layout and initializes it with the bytes in the provided
+     * source memory segment.
+     * @implSpec the default implementation for this method calls {@code this.allocate(elementLayout, elementCount)}.
+     * @param elementLayout the value to be set on the newly allocated memory block.
```

Comment:
This reads a bit strange. How is the value set 'on' the newly allocated memory block? I'd suggest following the flavor of the other allocateFrom overloads, e.g.

```suggestion
     * @param elementLayout the element layout of the array to be allocated.
```

Given that we have a pair of element layouts, I think it's reasonable to say that the result is an 'array'. (not a Java array though)